### ISSUE TITLE
Symbolic  differential support

### ISSUE THREAD

**Issue by oovm** [p-e-w/savage#6]
It would be nice if it could support symbolic differentiation, here is an implementation

https://github.com/egraphs-good/egg/blob/main/tests/math.rs

**Comment by p-e-w** [comment]
Indeed, support for symbolic differentiation and integration is planned.

And `egg` is a fascinating project, which was first pointed out to me on Reddit a few days ago. I absolutely intend to explore integrating it into Savage!
